Signal/src/ViewControllers/SignalsViewController.m

```diff
@@ -861,20 +862,28 @@ - (YapDatabaseConnection *)uiDatabaseConnection {
 }
 
 - (void)yapDatabaseModified:(NSNotification *)notification {
+    if (!self.shouldObserveDBModifications) {
+        return;
+    }
+
     NSArray *notifications  = [self.uiDatabaseConnection beginLongLivedReadTransaction];
-    NSArray *sectionChanges = nil;
-    NSArray *rowChanges     = nil;
+
+    if (![[self.uiDatabaseConnection ext:TSThreadDatabaseViewExtensionName] hasChangesForGroup:self.currentGrouping
```

Comment:
👍 

Will this solve the problem of requesting contact access when receiving your first message while the app is in the background?

I wonder if there's an edge case where we won't properly request contact access if you are viewing your archive when you receive your first message.